src/test/java/com/rackspace/ceres/app/web/QueryControllerTest.java

```diff
@@ -61,33 +60,35 @@ public void testQueryApi() {
   @Test
   public void testQueryApiWithAggregator() {
 
-    Map<String, String> queryTags = new HashMap<>();
-    queryTags.put("os", "linux");
-    queryTags.put("deployment", "dev");
-    queryTags.put("host", "h-1");
+    Map<String, String> queryTags = Map.of("os", "linux", "deployment", "dev", "host", "h-1");
 
-    Map<Instant, Double> values = new HashMap<>();
-    values.put(Instant.now(), 111.0);
+    Map<Instant, Double> values = Map.of(Instant.now(), 111.0);
 
-    List<QueryResult> queryResults = new ArrayList<>();
-    queryResults.add(new QueryResult().setMetricName("cpu-idle").setTags(queryTags).setTenant("t-1")
-        .setValues(values));
+    List<QueryResult> queryResults = List
+        .of(new QueryResult().setMetricName("cpu-idle").setTags(queryTags).setTenant("t-1")
+            .setValues(values));
 
     when(queryService.queryDownsampled(anyString(), anyString(), any(), any(), any(), any(), any()))
-        .thenReturn(Flux.just(queryResults.get(0)));
+        .thenReturn(Flux.fromIterable(queryResults));
 
     webTestClient.get()
         .uri(uriBuilder -> uriBuilder.path("/api/query")
             .queryParam("tenant", "t-1")
             .queryParam("metricName", "cpu-idle")
-            .queryParam("tag", "os=linux")
-            .queryParam("start", "1d-ago")
-            .queryParam("end", "1605117336")
+            .queryParam("tag", "os=linux,deployment=dev,host=h-1,")
```

Comment:
Does this test case succeed? `tag` is a repeated query parameter and the code splits off at the first equals sign

https://github.com/racker/ceres/blob/ec3bc6b05d817101fb9c272ba048e1e5d21df12a/src/main/java/com/rackspace/ceres/app/web/TagListConverter.java#L27

so I thought it turn your query into one tag with key `os` and value `linux,deployment=dev,host=h-1,`.

Perhaps Spring splits on the commas for us...which would be cool.

Reply:
The test case succeeded for me and yes spring does splits on the commas as the request param List<Strings> is having list as `["os=linux", "deployment=dev", "host=h-1]`

Reply:
Very nice! I'm glad you included that in the test.